### PR TITLE
chore: improvement to issue template

### PR DESCRIPTION
**Pull request by eladb**
1. Move description to the top, so it will be easier to skim through issues in mobile email clients.
2. Move "This is a feature request" to the bottom. It's not really needed since we have tags
3. Move "General Information" check boxes to "Other" at the end
4. Add "Use Case" to encourage people to describe WHY
5. Simplify and shorten instructions
6. Remove "Environment". It's usually not relevant for feature requests.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -4,43 +4,45 @@ about: Report a bug
...
 ### Environment
 
-  - **CDK CLI Version:** <!-- Output of `cdk version` -->
-  - **Module Version:** <!-- Version of the module in question -->
-  - **OS:** <!-- [all | Windows 10 | OSX Mojave | Ubuntu | etc... ] -->
-  - **Language:** <!-- [all | TypeScript | Java | Python ] etc... ] -->
+  - **CLI Version.     :**
+  - **Framework Version:**
+  - **OS               :**
+  - **Language.        :**
```
Comment (RomainMuller) [inline .github/ISSUE_TEMPLATE/bug.md]: Drop the `.`?

Comment (eladb) [reply .github/ISSUE_TEMPLATE/bug.md]: Done

File: .github/ISSUE_TEMPLATE/bug.md [hunk 2]
```diff
@@ -4,43 +4,45 @@ about: Report a bug
...
 
-If your minimal reprodiction is large, you may provide references to a [gist][1]
-or any other publicly viewable location.
+<!--
+minimal amount of code that causes the bug (if possible) or a reference:
 -->
 
-### Verbose Log
-<!--
-If the output is very large, you may store it in a [gist][1]. In any case,
```
Comment (NGL321) [inline .github/ISSUE_TEMPLATE/bug.md]: You dont want to reference Gist here?

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 3]
```diff
@@ -4,45 +4,39 @@ about: Request a new feature
...
--->
 
 ### Proposed Solution
-<!--
-Whenever relevant, describe how you would like the feature to be implemented.
-Include any documentation that can help understand your idea in very concrete
-ways, such as code examples that leverage your feature, captures of design
-diagrams, ...
+
+<!-- prototype/workaround/sketch/reference: -->
```
Comment (NGL321) [inline .github/ISSUE_TEMPLATE/feature-request.md]: What does this mean?

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 4]
```diff
@@ -4,45 +4,39 @@ about: Request a new feature
...
--->
-* [ ] :wave: I may be able to implement this feature request
-<!--
-Check the box below (with an X) if you think this feature might result in a
-breaking change (this requiring a major version bump when released). If unsure,
-please leave the box un-checked.
--->
-* [ ] :warning: This feature might incur a breaking change
 
-### Description
```
Comment (NGL321) [inline .github/ISSUE_TEMPLATE/feature-request.md]: If we are cutting down to one description section, I think "Description" makes more sense than "Proposed Solution"

File: .github/ISSUE_TEMPLATE/bug.md (hunk 2 above)
Comment (eladb) [reply .github/ISSUE_TEMPLATE/bug.md]: I haven't seen this as an issue. If we see this becoming an issue we can add a note.

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 4 above)
Comment (eladb) [reply .github/ISSUE_TEMPLATE/feature-request.md]: The main description is at the top of the issue (without a heading, to reduce clutter). The second is "proposed solution" with a remark that suggests to include a prototype/sample code.

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 3 above)
Comment (eladb) [reply .github/ISSUE_TEMPLATE/feature-request.md]: I'll improve the wording and remove the proposed solution header

**Comment by nmussy** [comment]
@eladb The [doc.md](https://github.com/aws/aws-cdk/blame/master/.github/ISSUE_TEMPLATE/doc.md#L4) template has the `feature-request` label, and no label to indicate that it's a `docs` issue. Was that intended or copypasta?

**Pull request merged**